src/status_im/contexts/settings/wallet/events.cljs

```diff
@@ -170,3 +200,36 @@
                :text  (:error error-data)}]]]})))
 
 (rf/reg-event-fx :wallet/import-keypair-by-seed-phrase-failed import-keypair-by-seed-phrase-failed)
+
+(defn import-keypair-by-private-key
+  [_ [{:keys [keypair-key-uid private-key password on-success on-error]}]]
+  {:fx [[:import-keypair-by-private-key
+         {:private-key private-key
+          :password    password
+          :on-success  (fn []
+                         (rf/dispatch [:wallet/make-keypairs-accounts-fully-operable
+                                       #{keypair-key-uid}])
+                         (cond
+                           (vector? on-success) (rf/dispatch (conj on-success))
+                           (fn? on-success)     (on-success)))
+          :on-error    (fn [error]
+                         (rf/dispatch [:wallet/import-keypair-by-private-key-failed error])
+                         (cond
+                           (vector? on-error) (rf/dispatch (conj on-error error))
+                           (fn? on-error)     (on-error error)))}]]})
```

Comment:
Just wondering why we need to dispatch an effect with two levels of promises in it where we can dispatch the `wallet/make-private-key-keypair-fully-operable` event directly. Sometimes, I felt we promisify code a lot. Sorry, It's really helpful in native module calls but for direct RPC calls, It's a bit overkill.

Reply:
This might be code left over from when I had originally composed to effects into one effect. But I recently refactored the logic to call the two effects separately, so we don't need this extra wrapper layer anymore I think 👍